csi.proto

```diff
@@ -802,6 +802,16 @@ message ListVolumesRequest {
 message ListVolumesResponse {
   message Entry {
     Volume volume = 1;
+
+    // A list of all `node_id` of nodes that the volume in this entry
+    // is controller published on.
+    // This is a REQUIRED field if the SP has the
+    // LIST_VOLUMES_PUBLISHED_NODES controller capability. If the SP
+    // does not have the LIST_VOLUMES_PUBLISHED_NODES capability this
+    // field MUST be empty.
+    // published_nodes MAY include volumes not published or
+    // created by the driver. The CO MUST be resilient to that
+    repeated string published_nodes = 2;
```

Comment:
The definition of REQUIRED from specs read:

> * A `repeated` or `map` field listed as `REQUIRED` MUST contain at least 1 element.

What happens when volume being listed is not published anywhere?

Reply:
amended to be optional with more verbose wording

Reply:
Okay, so if this is optional - why do we need to introduce a capability for this?

Reply:
So that if a driver doesn't implement it, we don't assume that the volume is not published

Reply:
ack.